server/user_account/views.py

```diff
@@ -1,13 +1,39 @@
+"""
+This file provides functionality for all the endpoints for interacting with user accounts
+"""
+
 from django.db.models import signals
 from django.contrib.auth.hashers import check_password
 from rest_framework import status, viewsets, generics
 from rest_framework.authtoken.models import Token
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes
 from .serializers import UserSerializer, LoginSerializer, ClientGridSerializer
 from .models import CustomUser
 
 
+# TODO: Remove this when registration view is updated and don't forget to remove the associated URL
+@api_view(['GET', 'PUT'])
+@permission_classes([IsAuthenticated])
+def get_employee(request, the_id):
+    """
+    Function for getting an employee and updating it
+    """
+    employee = CustomUser.objects.get(id=the_id)
+
+    if request.method == 'GET':
+        employee_serializer = ClientGridSerializer(employee)
+        return Response(employee_serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "PUT":
+        employee_serializer = ClientGridSerializer(employee, data=request.data)
+        if employee_serializer.is_valid():
+            employee_serializer.save()
+            return Response(employee_serializer.data, status=status.HTTP_200_OK)
+        return Response(employee_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Just to explain the future vision: The "GET" a specific employee would now be done through the RegistrationView which is now renamed to CustomUserView. The "PUT" will now be done throught the View that I just created called UpdateProfileView, although I will just add permissions to these two views to be not only accessible by the owner but also by the admin.

Reply:
Why do are we splitting up the views like that?
Wouldn't it just make more sense to have them all under a single class for CRUD operations?

Reply:
Well actually we could do that using the RetrieveUpdateAPIView as shown in the DRF doc:
<img width="557" alt="Screen Shot 2020-10-31 at 3 20 09 PM" src="https://user-images.githubusercontent.com/35696935/97788088-925ef480-1b8c-11eb-9553-fbb685999b27.png">

But what I did in the other pull request is that i just used an UpdateAPIView as shown in the DRF doc:
<img width="428" alt="Screen Shot 2020-10-31 at 3 21 06 PM" src="https://user-images.githubusercontent.com/35696935/97788106-b4587700-1b8c-11eb-922d-d613bf35eba0.png">

because the "GET" was already provided by the viewset so If was like why not use it. Its just a matter of preferance. 

Also another reason why i chose to use the viewset for the GET is because the serializer is returning everything that we need for the user:
![Screen Shot 2020-10-31 at 3 25 39 PM](https://user-images.githubusercontent.com/35696935/97788177-56785f00-1b8d-11eb-80b7-808cf6315cc1.png)

So the Update serializer is more restricted as shown below:
![Screen Shot 2020-10-31 at 3 26 37 PM](https://user-images.githubusercontent.com/35696935/97788191-7871e180-1b8d-11eb-9f23-d35fb72dda5b.png)

I won't allow the user to update any field that is not present here.

Let me know what do you think behind my reasoning.

Reply:
I see what you mean, but is there no way to set the serializers at a method level?
For example, 1 serializer can be used for get, and the next would be used for put and patch?

Also, another detail that I want to mention is that we should add more fields to the Update serializer. We should be able to modify email, role and is_active at the very least. We're going to be removing user_name, so that can be removed.

Reply:
So for the first point, We could do that but since the serializer_class is usually generalized for all of the actions on a class based view the only way to define it is to override the specific method responsible for put/patch to say that we want to use another serializer class. So i thought about it because i could have implemented all of that inside the registration viewset but i thought it would be a bit a mess since i'm already overriding things there for the create user.
Do you understand my choice?

For the second point, yes i can add those but if i'm a user would i want to be able to change my role etc.. I'm talking about the case of any user updating their profile. That's why i didn't have those.

Reply:
For the first point, I see why you did it, but have you tried this:
![get_serializer](https://user-images.githubusercontent.com/40705372/97789829-12d82200-1b9a-11eb-969d-a1bbead3fbb8.png)

It should help you provide some flexibility to what serializers you return. You can maybe set it up to that based on the action, you will return a specific serializer.

For the second point, I see what you mean, but maybe also the above pic can help you also modify the serializer based on the role?

Reply:
I will look into the first point, but can you provide the link i'm not able to find it on the DRF doc.
For the second point, once i will look into it we can discuss on the other PR this choice.

Reply:
Here's the link: [https://www.django-rest-framework.org/api-guide/generic-views/](https://www.django-rest-framework.org/api-guide/generic-views/)

It's in the generic-views part, but yeah let's take this over to PR #116